### PR TITLE
feat: add new private bucket for openedx-learning

### LINKED ISSUE #48: MinIO needs to handle private media assets
In Sumac, Open edX needs an additional bucket for the storage of private media assets. See: https://github.com/overhangio/tutor/pull/1124/

We will need to handle those assets in MinIO as well.

### PR DESCRIPTION
**Pull request by Faraz32123**
- add a new bucket to handle private media for openedx-learning.
- add changelog entry.
- close #48

### CODE REVIEW

File: tutorminio/templates/minio/tasks/minio/init.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 mc config host add minio http://minio:9000 {{ OPENEDX_AWS_ACCESS_KEY }} {{ OPENEDX_AWS_SECRET_ACCESS_KEY }} --api s3v4
-mc mb --ignore-existing minio/{{ MINIO_BUCKET_NAME }} minio/{{ MINIO_FILE_UPLOAD_BUCKET_NAME }} minio/{{ MINIO_VIDEO_UPLOAD_BUCKET_NAME }} minio/{{ MINIO_GRADES_BUCKET_NAME }}
+mc mb --ignore-existing minio/{{ MINIO_BUCKET_NAME }} minio/{{ MINIO_FILE_UPLOAD_BUCKET_NAME }} minio/{{ MINIO_VIDEO_UPLOAD_BUCKET_NAME }} minio/{{ MINIO_GRADES_BUCKET_NAME }} minio/{{ MINIO_OPENEDX_LEARNING_BUCKET_NAME }}
```
Comment (DawoudSheraz) [inline tutorminio/templates/minio/tasks/minio/init.sh:2]: Is this bucket private? Or is this only for storing private content?

Comment (Faraz32123) [reply tutorminio/templates/minio/tasks/minio/init.sh:2]: this is a private bucket to handle private media for openedx-learning.

**Review by DawoudSheraz — APPROVED** [review]

File: tutorminio/patches/openedx-common-settings [hunk 2]
```diff
@@ -3,6 +3,10 @@ VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_IMAGE_SETTINGS["STORA
 VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"].lstrip("/")
 GRADES_DOWNLOAD["STORAGE_KWARGS"] = {"location": GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"].lstrip("/")}
 GRADES_DOWNLOAD["STORAGE_KWARGS"]["bucket_name"] = "{{ MINIO_GRADES_BUCKET_NAME }}"
+OPENEDX_LEARNING["MEDIA"]["BACKEND"] = DEFAULT_FILE_STORAGE
+OPENEDX_LEARNING["MEDIA"]["OPTIONS"] = {
+    'bucket_name': "{{ MINIO_OPENEDX_LEARNING_BUCKET_NAME }}",
```
Comment (ormsbee) [inline tutorminio/patches/openedx-common-settings:8]: Does there need to be anything to force this bucket to be private? Or does that happen by default?

Comment (Faraz32123) [reply tutorminio/patches/openedx-common-settings:8]: buckets are by-default private in minio, we do have the policy to make the bucket public.

**Comment by Faraz32123** [comment]
tested this PR on local environment.
- checkout tutor branch to `regisb/meilisearch` as library_v2 requires `meilisearch`.
- enable minio plugin with this branch `feat/add_private_bucket_to_handle_private_media_for_openedx_learning` and run it's init job.
- make sure you have latest build for mfe(authoring) and openedx.
- create a library in `Libraries Beta` tab.
Screenshot 2024-10-30 at 8 06 00 PM

- add a new component(Text component) in it or can also copy text/html component from course and paste that in your library.
Screenshot 2024-10-30 at 8 12 10 PM

- click on this new component, go to Details -> Advanced details. Upload a picture there! 
Screenshot 2024-10-30 at 8 15 14 PM

- Now if u go to the MINIO_CONSOLE_HOST that would be `http://minio.local.openedx.io/login`. You'll see that the image is upload to the new private bucket named `openedxlearning`. To view the image, you'll have to download it.
Screenshot 2024-10-30 at 8 20 33 PM

**Comment by Faraz32123** [comment]
> tested this PR on local environment.

Note: In library_v2 testing, the code currently deduplicates content(images) within each library by hashing, preventing re-upload within the same library. If an image is uploaded, deleted from file storage or MinIO, and then re-uploaded in the same library, it will not upload again, resulting in an error while accessing the link to the image. To test uploading the same image, use a new library, as each library has its own isolated space/hash_folder.

File: tutorminio/patches/openedx-common-settings (hunk 2 above)
Comment (ormsbee) [reply tutorminio/patches/openedx-common-settings:8]: Okay, great. To be clear: we definitely want it to be private. Thank you.

Comment (Faraz32123) [reply tutorminio/patches/openedx-common-settings:8]: Screenshot 2024-10-31 at 11 24 02 AM

**Review by DawoudSheraz — APPROVED** [review]
Tested it using sumac sandbox as MINIO_HOST, the bucket is working and the files are getting uploaded from libraries v2 interface.

**Review by regisb — APPROVED** [review]

**Comment by Faraz32123** [comment]
This is tested across dev, local and k8s envs.

**Pull request merged**